dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/ProgramBuilderSpv.java

```diff
@@ -294,10 +296,23 @@ public Expression addExpression(String name, Expression value) {
         if (types.containsKey(name) || expressions.containsKey(name)) {
             throw new ParsingException("Duplicated definition '%s'", name);
         }
+        if (preThreadDecoration.containsKey(name)) {
```

Comment:
I would probably create another expression transformer for global variables (or alter the existing one) to keep all decorations at the same place, but your solution should work fine too. Let keep it this way for now, we can refactor it later if we have time.

Reply:
Okay